pkg/registry/apis/provisioning/jobs/pullrequest/preview.go

```diff
@@ -138,54 +108,59 @@ func (p *previewer) previewURL(u *url.URL, repoName, ref, filePath, pullRequestU
 // Preview creates a preview for a single file change
 func (p *previewer) Preview(
 	ctx context.Context,
-	f repository.VersionedFileChange,
-	namespace string,
-	repoName string,
-	base string,
-	ref string,
+	f *resources.ParsedResource,
 	pullRequestURL string,
 	generatePreview bool,
 ) (resourcePreview, error) {
+	namespace := f.Obj.GetNamespace()
 	baseURL, err := url.Parse(p.urlProvider(namespace))
 	if err != nil {
 		return resourcePreview{}, fmt.Errorf("error parsing base url: %w", err)
 	}
 
+	if f.GVK.Kind != "Dashboard" {
+		return resourcePreview{}, fmt.Errorf("only dashboards are supported")
+	}
+
+	err = f.DryRun(ctx)
```

Comment:
why do we need a dry run?

Reply:
This tells us what will happen when we apply the change -- and in particualr if the value is valid or not.

This leads to the bigger change I want to do... this function is not really a "preview" it is the "generate data we need to create a useful message" function

Reply:
it also lets us know if we are doign an update or create (the git status is just about the file, not about the referenced k8s name in the file)